modular_skyrat/modules/gun_cargo/code/datums/cargo_pack.dm

```diff
@@ -0,0 +1,16 @@
+/datum/supply_pack/armament
+
+/datum/supply_pack/armament/generate(atom/A, datum/bank_account/paying_account)
+	. = ..()
+	/*
+	var/obj/structure/container = .
+	for(var/obj/item/gun_maybe in container.contents)
+		if(!istype(gun_maybe, /obj/item/gun))
+			continue
+		var/obj/item/gun/gun_actually = gun_maybe
```

Comment:
```suggestion
	for(var/obj/item/gun_actually in container.contents)
```

You may also want to rename the variable to "gun" or "firearm", but it's entirely optional.